storagenode/piecestore/endpoint.go

```diff
@@ -506,10 +506,41 @@ func (endpoint *Endpoint) Download(stream pb.Piecestore_DownloadServer) (err err
 			currentOffset += chunkSize
 			unsentAmount -= chunkSize
 		}
-
 		return nil
 	})
 
+	//for repair traffic, also send along the PieceHash and OrderLimit for validation
+	if message.Limit.Action == pb.PieceAction_GET_REPAIR {
+		var orderLimit pb.OrderLimit
+		var pieceHash pb.PieceHash
+		//GetPieceHeader return BadFormatVersion err if v0
+		if header, err := pieceReader.GetPieceHeader(); err == nil {
+			//v1+ stores this information in the file
+			orderLimit = header.OrderLimit
+			pieceHash = pb.PieceHash{
+				PieceId:   limit.PieceId,
+				Hash:      header.GetHash(),
+				PieceSize: pieceReader.Size(),
+				Timestamp: header.GetCreationTime(),
+				Signature: header.GetSignature(),
+			}
+		} else if pieces.BadFormatVersion.Has(err) {
+			//v0 stores this information in SQL
+			info, err := endpoint.store.GetV0PieceInfoDB().Get(ctx, limit.SatelliteId, limit.PieceId)
+			if err != nil {
+				return status.Error(codes.Internal, err.Error())
+			}
+			orderLimit = *info.OrderLimit
```

Comment:
Do we need this back and forth `interface/value` conversion and additional allocation(not sure)?